src/test/java/com/yelp/nrtsearch/server/config/ScriptCacheConfigTest.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2021 Yelp Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yelp.nrtsearch.server.config;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.ByteArrayInputStream;
+import org.junit.Test;
+
+public class ScriptCacheConfigTest {
+  private static ScriptCacheConfig getConfig(String configFile) {
+    return ScriptCacheConfig.fromConfig(
+        new YamlConfigReader(new ByteArrayInputStream(configFile.getBytes())));
+  }
+
+  @Test
+  public void testDefault() {
+    String configFile = "nodeName: \"lucene_server_foo\"";
+    ScriptCacheConfig config = getConfig(configFile);
+    assertEquals(ScriptCacheConfig.DEFAULT_CONCURRENCY_LEVEL, config.getConcurrencyLevel());
+    assertEquals(ScriptCacheConfig.DEFAULT_SIZE, config.getMaximumSize());
+    assertEquals(ScriptCacheConfig.DEFAULT_EXPIRATION_TIME, config.getExpirationTime());
+  }
+
+  @Test
+  public void testConfig() {
+    String configFile =
+        String.join(
+            "\n",
+            "nodeName: \"lucene_server_foo\"",
+            "ScriptCacheConfig:",
+            "  concurrencyLevel: 4",
+            "  maximumSize: 1000",
+            "  expirationTime: 1");
```

Comment:
You will probably want to use different values than the default here, to make sure they are really being used.